apps/system/test/marionette/lib/system.js

```diff
@@ -363,6 +380,34 @@ System.prototype = {
     this.waitForActivityMenu();
   },
 
+  saveImage: function() {
+    this.client.switchToFrame();
+    this.client.waitFor(function() {
```

Comment:
nit: no need to do this waitFor, `this.appChromeContextMenu.displayed()` internally does a `waitForElement` on the context menu